pkg/util/util.go

```diff
@@ -1614,3 +1627,287 @@ func GetAvailabeKubeCli() string {
+		return ns.Spec.IBMCos.TargetBucket, nil
+	case nbv1.NSStoreTypeAzureBlob:
+		return ns.Spec.AzureBlob.TargetBlobContainer, nil
+	case nbv1.NSStoreTypeNSFS:
+		return "", nil
+	default:
+		return "", fmt.Errorf("failed to ger namespacestore %q target bucket", ns.Name)
+	}
+}
+
+// GetSecretFromSecretReference search and retruns a secret obj from a provided secret reference
+func GetSecretFromSecretReference(secretRef *corev1.SecretReference) (*corev1.Secret, error) {
+	if secretRef == nil || secretRef.Name == "" {
+		return nil, nil
+	}
+
+	o := KubeObject(bundle.File_deploy_internal_secret_empty_yaml)
+	secret := o.(*corev1.Secret)
+
+	secret.Name = secretRef.Name
+	secret.Namespace = secretRef.Namespace
+
+	if !KubeCheck(secret) {
+		return nil, fmt.Errorf(`❌ Could not get Secret %q in namespace %q`, secret.Name, secret.Namespace)
+	}
+
+	return secret, nil
+}
+
+// CheckForIdenticalSecretsCreds search and returns a secret name with identical credentials in the provided secret
+// the credentials to compare stored in mandatoryProp
+func CheckForIdenticalSecretsCreds(secret *corev1.Secret, mandatoryProp []string) *corev1.Secret {
+	if secret == nil {
+		return nil
+	}
+	nsList := &nbv1.NamespaceStoreList{
+		TypeMeta: metav1.TypeMeta{Kind: "NamespaceStoreList"},
+	}
+	bsList := &nbv1.BackingStoreList{
+		TypeMeta: metav1.TypeMeta{Kind: "BackingStoreList"},
+	}
+	KubeList(bsList, &client.ListOptions{Namespace: secret.Namespace})
+	KubeList(nsList, &client.ListOptions{Namespace: secret.Namespace})
+
+	for _, bs := range bsList.Items {
+		if bs.Spec.Type != nbv1.StoreTypePVPool {
+			secretRef, err := GetBackingStoreSecret(&bs)
+			if err != nil {
+				log.Errorf("%s", err)
+			}
+			if secretRef != nil {
+				usedSecret, err := GetSecretFromSecretReference(secretRef)
+				if err != nil {
+					log.Errorf("%s", err)
+				}
+				if usedSecret != nil && usedSecret.Name != secret.Name {
+					found := true
+					for _, key := range mandatoryProp {
+						found = found && usedSecret.StringData[key] == secret.StringData[key]
+					}
+					if found {
+						return usedSecret
+					}
+				}
+			}
+		}
+	}
+
+	for _, ns := range nsList.Items {
+		if ns.Spec.Type != nbv1.NSStoreTypeNSFS {
+			secretRef, err := GetNamespaceStoreSecret(&ns)
+			if err != nil {
+				log.Errorf("%s", err)
+			}
+			if secretRef != nil {
+				usedSecret, err := GetSecretFromSecretReference(secretRef)
+				if err != nil {
+					log.Errorf("%s", err)
+				}
+				if usedSecret != nil && usedSecret.Name != secret.Name {
+					found := true
+					for _, key := range mandatoryProp {
+						found = found && usedSecret.StringData[key] == secret.StringData[key]
+					}
+					if found {
+						return usedSecret
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// SetOwnerReference setting a owner reference of owner to dependent metadata with the field of blockOwnerDeletion: true
+// controllerutil.SetOwnerReference is doing the same thing but without blockOwnerDeletion: true
```

Comment:
please add this link https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L96
to your comment so it'll be better explained where it is taken from.

Also, I think that the comment https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L59 is super important in order to understand the logic of the owner refernece and why we can not put conroller reference on the secret (explains the backingstore and namespacestore controller changes) , please add it there

Reply:
Good idea